src/native/libs/System.Native/pal_uid.c

```diff
@@ -281,24 +280,27 @@ char* SystemNative_GetGroupName(uint32_t gid)
 #else
     // Platforms like Android API level < 24 do not have getgrgid_r available
     // Android API level >= 24 do have getgrgid_r available but call getgrgid internally anyway
-    result = getgrgid(gid);
+    struct group* result = getgrgid(gid);
```

Comment:
The only way this can return something that we don't need to free (without a memory leak) is by using static data.
So, we're sure this function is not thread-safe.

It would be good to add a mutex here. And same for `getpwuid`.

Reply:
Done.